torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5902,7 +5910,7 @@ def _make_data_dependent_error(
         unhinted_expr: sympy.Basic,
         *,
         size_oblivious_result: Optional[sympy.Basic] = None,
-    ) -> GuardOnDataDependentSymNode:
+    ) -> Union[GuardOnDataDependentSymNode, GuardOnDataDependentSymNodeStrict]:
```

Comment:
I wonder if it make sense to just avoid changugn the error to user error here
```
 def evaluate_expr(self, output_graph=None):
        try:
            return guard_scalar(self.sym_num)
        except GuardOnDataDependentSymNode as e:
            raise UserError(  # noqa: B904
                UserErrorType.ANTI_PATTERN,
                f"Consider annotating your code using torch._check*(). {str(e)}",
                case_name="constrain_as_size_example",
            )

```
instead of introducing a new error type